nssp_demo/forecast_state.py

```diff
@@ -49,26 +49,41 @@ def main(
     disease: str,
     report_date: str,
     state: str,
-    nssp_data_dir: Path | str,
+    facility_level_nssp_data_dir: Path | str,
+    state_level_nssp_data_dir: Path | str,
     param_data_dir: Path | str,
     output_data_dir: Path | str,
     n_training_days: int,
     n_forecast_days: int,
     n_chains: int,
     n_warmup: int,
     n_samples: int,
-    last_training_date: str,
+    last_training_date: str = None,
 ):
     logging.basicConfig(level=logging.INFO)
     logger = logging.getLogger(__name__)
 
+    available_facility_level_reports = [
+        datetime.strptime(f.stem, "%Y-%m-%d").date()
+        for f in Path(facility_level_nssp_data_dir).glob("*.parquet")
+    ]
+    available_state_level_reports = [
+        datetime.strptime(f.stem, "%Y-%m-%d").date()
+        for f in Path(state_level_nssp_data_dir).glob("*.parquet")
+    ]
+
     if report_date == "latest":
-        report_date = max(
-            f.stem for f in Path(nssp_data_dir).glob("*.parquet")
-        )
-    report_date = datetime.strptime(report_date, "%Y-%m-%d").date()
+        report_date = max(available_facility_level_reports)
+    else:
+        report_date = datetime.strptime(report_date, "%Y-%m-%d").date()
+
+    if report_date in available_state_level_reports:
+        state_report_date = report_date
+    else:
+        state_report_date = max(available_state_level_reports)
```

Comment:
To use the latest state data when hybridizing, but I see now that a more correct thing would be 

```suggestion
    elif report_date > max(available_state_level_reports):
        state_report_date = max(available_state_level_reports)
     else:
        state_report_date = None
```
And then error if neither the facility-level nor the state level has a sufficiently early recorded `report_date`